include/openssl/arm_arch.h

```diff
@@ -78,6 +78,56 @@
 // ARMV8_SHA512 indicates support for hardware SHA-512 instructions.
 #define ARMV8_SHA512 (1 << 6)
 
+// ARMV8_SHA3 indicates support for hardware SHA-3 instructions including EOR3.
+#define ARMV8_SHA3  (1 << 11)
+
+// ARMV8_AES_GCM_UNROLL8 indicates that the microarchitecture
+// support the 8x unrolling function of GCM due to a wider pipeline.
+#define ARMV8_AES_GCM_UNROLL8 (1 << 12)
```

Comment:
I don't particularly like mixing CPU capabilities and types of implementations in the CPU cap variable. All the other flags define specific features of the CPU, but `ARMV8_AES_GCM_UNROLL8` is not really a feature. I'd rather have, for example, `ARMV8_APPLE_M1` flag. Then we can determine when to run the code based on `ARMV8_SHA3` and  `ARMV8_APPLE_M1` flags presence in the capability variable.

Reply:
I hear you, dkostic. This discussion came up in the OpenSSL PR associated with this commit and concluded with the following https://github.com/openssl/openssl/pull/15916#discussion_r784020203:
> I still would recommend that there are bits in the OPENSSL_armcap
environment variable.
This is request is mostly about testability, since this switches code paths on or off dependent
on the CPU micro-architecture, but why shouldn't we have the ability to test other code paths
that may be non-optimal for the used CPU?
(unfortunately the OPENSSL_arm_midr and OPENSSL_armv8_rsa_neonized made their
way into 3.0, but I would rather like them to be converted to OPENSSL_armcap_P bits as well)

The comments prior to that express the different point of views. This code is not only for M1 but for any micro-architecture that has a wide AES/PMULL pipeline AND (any architecture that) has the SHA3 extension and

Reply:
I don't think the argument in the comment you linked to applies to our case. My point is that `wide AES/PMULL pipeline` and `SHA3 extension` are CPU capabilities so we should encode those in the capabilities variable, and then we can decide which code to run in runtime based on the value of those two bits. I don't see how this hinders testing capabilities because afaik we can still set the capabilities variable externally by defining appropriate environment variable.

Reply:
Done, changed as discussed. See https://github.com/aws/aws-lc/pull/721/commits/8d3bb74a09f14c66c760e0afaf1d56ecbe6c20ae